flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/table.scala

```diff
@@ -785,13 +793,19 @@ class Table(
     * will be processed by a single operator.
     *
     * @param groupWindow group-window that specifies how elements are grouped.
-    * @return A windowed table.
     */
-  def window(groupWindow: GroupWindow): GroupWindowedTable = {
+  def window(groupWindow: GroupWindow): Table = {
     if (tableEnv.isInstanceOf[BatchTableEnvironment]) {
       throw new ValidationException(s"Windows on batch tables are currently not supported.")
     }
-    new GroupWindowedTable(this, Seq(), groupWindow)
+    if (None == groupWindow.name) {
+      throw new ValidationException("An alias must be specified for the window.")
+    }
+    if (windowPool.contains(groupWindow.name.get)) {
+      throw new ValidationException("The window alias can not be duplicated.")
```

Comment:
s/"The window alias can not be duplicated."/"The same window alias has been defined"/